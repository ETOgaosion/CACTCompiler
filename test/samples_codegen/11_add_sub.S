iobuff:
	.zero 80
	.equ	ImmValue_2, 0x400b333333333333

	.equ	ImmValue_3, 0x4000cccccccccccd

	.equ	ImmValue_1, 0x40133333

	.equ	ImmValue_0, 0x404ccccd

.text
.align 1
.globl	main
.type	main, @function
main:
	sd	ra, -8(sp)
	mv	t0, zero
	sw	t0, -40(sp)
	mv	t0, zero
	sw	t0, -36(sp)
	mv	t0, zero
	sw	t0, -32(sp)
	mv	t0, zero
	sw	t0, -28(sp)
	addi	sp, sp, -136
	call	get_int
	addi	sp, sp, 136
	sw	a0, -104(sp)
	lw	t0, -104(sp)
	sw	t0, -40(sp)
	lw	t0, -40(sp)
	addi	t1, zero, 3
	subw	t2, t0, t1
	sw	t2, -96(sp)
	lw	t0, -96(sp)
	sw	t0, -36(sp)
	addi	sp, sp, -136
	call	get_int
	addi	sp, sp, 136
	sw	a0, -92(sp)
	lw	t0, -92(sp)
	sw	t0, -32(sp)
	lw	t1, -32(sp)
	addi	t0, t1, 2
	sw	t0, -84(sp)
	lw	t0, -84(sp)
	sw	t0, -28(sp)
	lw	a0, -36(sp)
	addi	sp, sp, -136
	call	print_int
	addi	sp, sp, 136
	lw	a0, -28(sp)
	addi	sp, sp, -136
	call	print_int
	addi	sp, sp, 136
	fmv.w.x	ft0, zero
	fsw	ft0, -56(sp)
	fmv.w.x	ft0, zero
	fsw	ft0, -52(sp)
	fmv.w.x	ft0, zero
	fsw	ft0, -64(sp)
	fmv.w.x	ft0, zero
	fsw	ft0, -60(sp)
	addi	sp, sp, -136
	call	get_float
	addi	sp, sp, 136
	fsw	fa0, -100(sp)
	flw	ft0, -100(sp)
	fsw	ft0, -56(sp)
	flw	ft0, -56(sp)
	li	t0, ImmValue_0
	fmv.w.x	ft0, t0
	fsub.s	ft0, ft0, ft0
	fsw	ft0, -88(sp)
	flw	ft0, -88(sp)
	fsw	ft0, -52(sp)
	addi	sp, sp, -136
	call	get_float
	addi	sp, sp, 136
	fsw	fa0, -108(sp)
	flw	ft0, -108(sp)
	fsw	ft0, -64(sp)
	flw	ft0, -64(sp)
	li	t0, ImmValue_1
	fmv.w.x	ft0, t0
	fadd.s	ft0, ft0, ft0
	fsw	ft0, -120(sp)
	flw	ft0, -120(sp)
	fsw	ft0, -60(sp)
	flw	fa0, -52(sp)
	addi	sp, sp, -136
	call	print_float
	addi	sp, sp, 136
	flw	fa0, -60(sp)
	addi	sp, sp, -136
	call	print_float
	addi	sp, sp, 136
	fmv.d.x	ft0, zero
	fsd	ft0, -72(sp)
	fmv.d.x	ft0, zero
	fsd	ft0, -48(sp)
	fmv.d.x	ft0, zero
	fsd	ft0, -24(sp)
	fmv.d.x	ft0, zero
	fsd	ft0, -16(sp)
	addi	sp, sp, -136
	call	get_double
	addi	sp, sp, 136
	fsd	fa0, -128(sp)
	fld	ft0, -128(sp)
	fsd	ft0, -72(sp)
	fld	ft0, -72(sp)
	li	t0, ImmValue_2
	fmv.d.x	ft0, t0
	fadd.d	ft0, ft0, ft0
	fsd	ft0, -116(sp)
	fld	ft0, -116(sp)
	fsd	ft0, -48(sp)
	addi	sp, sp, -136
	call	get_double
	addi	sp, sp, 136
	fsd	fa0, -136(sp)
	fld	ft0, -136(sp)
	fsd	ft0, -24(sp)
	fld	ft0, -24(sp)
	li	t0, ImmValue_3
	fmv.d.x	ft0, t0
	fadd.d	ft0, ft0, ft0
	fsd	ft0, -80(sp)
	fld	ft0, -80(sp)
	fsd	ft0, -16(sp)
	fld	fa0, -48(sp)
	addi	sp, sp, -136
	call	print_double
	addi	sp, sp, 136
	fld	fa0, -16(sp)
	addi	sp, sp, -136
	call	print_double
	addi	sp, sp, 136
	mv	a0, zero
	ld	ra, -8(sp)
	ret
