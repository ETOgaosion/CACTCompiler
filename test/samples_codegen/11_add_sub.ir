============ global var ================
============ imm value ================
ImmValue_2: 3.4
ImmValue_3: 2.1
ImmValue_1: 2.3f
ImmValue_0: 3.2f
============ functions ================
----------- funcDef -----------
main() => 2:
----------- paramVars -----------
----------- localVars -----------
dd := symbol type: 1; data type: 4; initValue: 0

dc := symbol type: 1; data type: 4; initValue: 0

d := symbol type: 1; data type: 2; initValue: 0

c := symbol type: 1; data type: 2; initValue: 0

b := symbol type: 1; data type: 2; initValue: 0

a := symbol type: 1; data type: 2; initValue: 0

db := symbol type: 1; data type: 4; initValue: 0

fb := symbol type: 1; data type: 3; initValue: 0

fa := symbol type: 1; data type: 3; initValue: 0

fd := symbol type: 1; data type: 3; initValue: 0

fc := symbol type: 1; data type: 3; initValue: 0

da := symbol type: 1; data type: 4; initValue: 0

----------- tempVars ------------
t_22:= symbol type: TEMP VAR; data type: 4
t_6:= symbol type: TEMP VAR; data type: 2
t_10:= symbol type: TEMP VAR; data type: 3
t_4:= symbol type: TEMP VAR; data type: 2
t_2:= symbol type: TEMP VAR; data type: 2
t_8:= symbol type: TEMP VAR; data type: 3
t_0:= symbol type: TEMP VAR; data type: 2
t_12:= symbol type: TEMP VAR; data type: 3
t_18:= symbol type: TEMP VAR; data type: 4
t_14:= symbol type: TEMP VAR; data type: 3
t_16:= symbol type: TEMP VAR; data type: 4
t_20:= symbol type: TEMP VAR; data type: 4
----------- Codes ------------
main:
	a = 0;
	b = 0;
	c = 0;
	d = 0;
	call get_int; 
	t_0 =  return ;
	t_1 <= a;
	t_1 = t_0;
	t_2 = t_1 - 3;
	t_3 <= b;
	t_3 = t_2;
	call get_int; 
	t_4 =  return ;
	t_5 <= c;
	t_5 = t_4;
	t_6 = t_5 + 2;
	t_7 <= d;
	t_7 = t_6;
	addparam t_3;
	call print_int; 
	addparam t_7;
	call print_int; 
	fa = 0;
	fb = 0;
	fc = 0;
	fd = 0;
	call get_float; 
	t_8 =  return ;
	t_9 <= fa;
	t_9 = t_8;
	t_10 = t_9 - ImmValue_0: 3.2f;
	t_11 <= fb;
	t_11 = t_10;
	call get_float; 
	t_12 =  return ;
	t_13 <= fc;
	t_13 = t_12;
	t_14 = t_13 + ImmValue_1: 2.3f;
	t_15 <= fd;
	t_15 = t_14;
	addparam t_11;
	call print_float; 
	addparam t_15;
	call print_float; 
	da = 0;
	db = 0;
	dc = 0;
	dd = 0;
	call get_double; 
	t_16 =  return ;
	t_17 <= da;
	t_17 = t_16;
	t_18 = t_17 - ImmValue_2: 3.4;
	t_19 <= db;
	t_19 = t_18;
	call get_double; 
	t_20 =  return ;
	t_21 <= dc;
	t_21 = t_20;
	t_22 = t_21 + ImmValue_3: 2.1;
	t_23 <= dd;
	t_23 = t_22;
	addparam t_19;
	call print_double; 
	addparam t_23;
	call print_double; 
	return 0;

