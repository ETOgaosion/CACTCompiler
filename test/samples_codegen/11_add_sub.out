bbl loader
please enter an int number:
please enter an int number:
-2
4
please enter an float number:
please enter an float number:
1.300000
8.800000
please enter an double number:
please enter an double number:
8.743000
125.400000
